std/math.d

```diff
@@ -1150,106 +1204,180 @@ real atan(real x) @safe pure nothrow @nogc
 {
     version(InlineAsm_X86_Any)
     {
-        return atan2(x, 1.0L);
+        if (!__ctfe)
+            return atan2Asm(x, 1.0L);
+    }
+    return atanImpl(x);
```

Comment:
Perhaps all functions could be split up this way between asm and impl. `floor`/`ceil` are two others where this could be applied too.

I forsee an opportunity to split std.math perhaps in the same way bigint is divided into generic and asm modules. Maybe even add ldc and gdc asm implementations also. ;-)

Reply:
In the long run, I'd prefer std.math being split up into a package, with a dedicated module per function family (atan, exp, etc.). Editing that huge file would be much easier then.

Reply:
Keeping similar things (e.g. docs) in files in sync would become more difficult and switching between files is quite an overhead. Might be a better idea to outsource some heavy implementations to `std/internal/math/` if module size is a concern.